libcudacxx/include/cuda/__iterator/zip_iterator.h

```diff
@@ -182,24 +259,50 @@ class zip_iterator : public __zv_iter_category_base<_Iterators...>
     return reference{*__iters...};
   }
 
+  //! @brief Dereferences the @c zip_iterator
+  //! @returns A tuple of references obtained by referencing every stored iterator
   [[nodiscard]] _CCCL_API constexpr auto operator*() const
     noexcept(noexcept(::cuda::std::apply(__zip_op_star, __current_)))
   {
     return ::cuda::std::apply(__zip_op_star, __current_);
   }
 
+  struct __zip_op_index
+  {
+    difference_type __n;
+
+    _CCCL_EXEC_CHECK_DISABLE
+    [[nodiscard]] _CCCL_API constexpr reference operator()(const _Iterators&... __iters) const
+      noexcept(noexcept(reference{__iters[::cuda::std::iter_difference_t<_Iterators>(__n)]...}))
+    {
+      return reference{__iters[::cuda::std::iter_difference_t<_Iterators>(__n)]...};
+    }
+  };
+
+  //! @brief Subscripts the @c zip_iterator with an additional offset
+  //! @param __n The additional offset
+  //! @returns A tuple of references obtained by subscripting every stored iterator
+  _CCCL_TEMPLATE(class _Constraints = __zip_iter_constraints<_Iterators...>)
+  _CCCL_REQUIRES(_Constraints::__all_random_access)
+  _CCCL_API constexpr auto operator[](difference_type __n) const
+  {
+    return ::cuda::std::apply(__zip_op_index{__n}, __current_);
+  }
+
   _CCCL_EXEC_CHECK_DISABLE
   _CCCL_API static constexpr void __zip_op_increment(_Iterators&... __iters) noexcept(noexcept(((void) ++__iters, ...)))
   {
     ((void) ++__iters, ...);
   }
 
+  //! @brief Increments all stored iterators
   _CCCL_API constexpr zip_iterator& operator++() noexcept(noexcept(::cuda::std::apply(__zip_op_increment, __current_)))
   {
     ::cuda::std::apply(__zip_op_increment, __current_);
     return *this;
   }
 
+  //! @brief Increments all stored iterators
```

Comment:
```suggestion
  //! @brief Increments all stored iterators
  //! @returns A copy of the original @c zip_iterator if possible
```
Arguably `x++` and `x--` should document the return value for all fancy iterators.